### DISCUSSION TITLE
web services over abs proxy but dynamic port?

### DISCUSSION THREAD

**Discussion by Blakeinstein** [coder/code-server#3872] [Q&A]
I am running into the problem of being able to host my dev server on dynamic ports.

As per this https://coder.com/docs/code-server/v3.11.0/guide#proxying-to-create-a-react-app 
I can proxy my content to that url but I will never know the port.

**Comment by jsjoeio** [comment]
Hmm... 🤔 In your project, which piece is in charge of setting the port? 

i.e. I know in CRA, you can use the PORT environment variable ([source](https://stackoverflow.com/a/41770848))

**Reply by Blakeinstein** [reply]
While yes, it is highly unlikely that the port would change. But I would like to be able to work with front end projects, with minimal config. [This](https://github.com/Blakeinstein/codeserver-live) is the end project I am aiming to build. If it was possible to show simple browser, on demand from vscode itself... most of my problems would be solve

**Reply by jsjoeio** [reply]
Hmmm... okay reiterating the problem in my own words, we want to know how to access web services over the absolute proxy using a dynamic port?

It's almost like you need middleware to handle the dynamic port, right? Like a layer in between code-server's abs proxy and your web services 🤔

Maybe @code-asher has a good idea how to solve this.

**Reply by code-asher** [reply]
I think the built-in extension we aim to build should resolve this. It will look for ports with something running on them then show that in VS Code so they can be clicked and opened.

**Reply by code-asher** [reply]
https://github.com/cdr/code-server/issues/1944

**Reply by Blakeinstein** [reply]
Can we integrate this with the simple browser inside code-server? and somehow pipe that through code-server?

**Reply by code-asher** [reply]
Yeah that seems like a cool feature.
